wherobots/db/cursor.py

```diff
@@ -54,20 +61,21 @@ def __get_results(self) -> Optional[List[Tuple[Any, ...]]]:
         if isinstance(result, DatabaseError):
             raise result
 
-        self.__rowcount = len(result)
-        self.__results = result
-        if not result.empty:
+        columns, column_types, rows = result
+        self.__rowcount = len(rows)
+        self.__results = rows
+        if rows:
             self.__description = [
                 (
                     col_name,  # name
-                    _TYPE_MAP.get(str(result[col_name].dtype), "STRING"),  # type_code
+                    _TYPE_MAP.get(str(column_types[i]), "STRING"),  # type_code
                     None,  # display_size
-                    result[col_name].memory_usage(),  # internal_size
+                    None,  # internal_size
                     None,  # precision
                     None,  # scale
                     True,  # null_ok; Assuming all columns can accept NULL values
                 )
-                for col_name in result.columns
+                for i, col_name in enumerate(columns)
```

Comment:
Use https://docs.python.org/3/library/functions.html#zip to avoid jumping hoops with an index (it's much nicer to read, and also more efficient):

```python
self.__description = [
  (
    col_name,
    _TYPE_MAP.get(col_type, 'STRING'),
    ...
  )
  for (col_name, col_type) in zip(columns, column_types)
]
```